ts/sagas/wallet.ts

```diff
@@ -298,7 +307,36 @@ function* startOrResumeAddCreditCardSaga(
       const maybeAddedWallet = updatedWallets.find(
         _ => _.idWallet === idWallet
       );
+      // if the new method has been added
       if (maybeAddedWallet !== undefined) {
+        const bpdEnroll: ReturnType<typeof bpdEnabledSelector> = yield select(
+          bpdEnabledSelector
+        );
+        // check if the new method is compliant with bpd
+        if (bpdEnabled && maybeAddedWallet.v2) {
+          const hasBpdFeature = hasFunctionEnabled(
+            maybeAddedWallet.v2,
+            EnableableFunctionsTypeEnum.BPD
+          );
+          // if the method is bpd compliant check if we have info about bpd activation
+          if (hasBpdFeature && isReady(bpdEnroll)) {
+            // if bdp is active navigate to a screen where it asked to enroll that method in bpd
+            // otherwise navigate to a screen where is asked to join bpd
+            if (bpdEnroll.value) {
+              yield put(
+                navigateToActivateBpdOnNewMethod({
+                  newAddedMethods: [maybeAddedWallet.v2]
+                })
+              );
+            } else {
+              yield put(navigateToSuggestBpdActivation());
+            }
+            // remove these screen from the navigation stack: method choice, credit card form, credit card resume
+            yield put(navigationHistoryPop(3));
+            return;
```

Comment:
This could be a problem because we have a story that ask us to add a credit card without the method choise https://www.pivotaltracker.com/story/show/175757212
We could accept this implementation but we should freeze the credit card onboarding workflow and reject this request.